### PR TITLE
Changes the scope component to be appliable to other items and not just guns. Binocs and the Curator's mothic cap use it.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
I've done some changes to the scope component so that it can be applied to items other than guns.

Binoculars now use it instead of changing the player view size, and lets them move around while the zoom is on at the cost of a moderate slowdown. This is both more comfortable, cooler and less laggier than the previous method. Take in mind it still takes both hands to use binoculars, so there's no much room for exploitation anyway.

This PR also makes the mothic softcap from the curator kit special: it too has the scope component, albeit of a much shorter range, because it's probably the least interesting kit of them all right now and has nothing that the curator cannot get by walking in the aux base construction room, except for a pair of pockets in the mothic coat, though that's superseeded by the Trailwarden kit's saddlepack. It's pretty fitting if you ask since the softcap has some goofy-looking googles drawn on it.

Oh, items that aren't guns also have a different mouse tracker icon of two circles joined together, instead of a reticle.

## Why It's Good For The Game
I'm aiming to make binoculars and the Mothic Pioneer kit more enjoyable.

## Changelog
:cl:
add: Reworked Binoculars to function like a scope from long-range rifles. They also can be used while moving, albeit that'll slow you down significantly.
add: Authentic mothic softcaps can be found in the "Mothic Pioneer" curator kit, which googles can also used to look into the distance. The date on the boxed kit has been updated to match "muh lore".
qol: Using a scope (and now binocs or moth cap) won't hide your mouse pointer away now, so that you won't slip out of the game screen and close the game tab by accident.
/:cl:

### CODE REVIEW

**Comment by Helg2** [comment]
Was the component made usable or no? From what I remember it's mouce sensivity was very uncomfortable to use.

**Comment by Ghommie** [comment]
> Was the component made usable or no? From what I remember it's mouce sensivity was very uncomfortable to use.

I think that's also partly due to the mouse pointer being invisible while the zoom is enabled, which gets quite awful when it's moved outside of the game screen. I can easily change it so that the pointer remains visible while it's on, but smoothing out the scope would be an extra bit of effort I'm not too willing to spend on this PR.

**Comment by Mickyan** [comment]
I had some plans to add something unique to the pioneer bundle since I struggled to think of anything back when I added it, but this is more interesting, I approve!

While you're in there could you do me a favor and change the date of the bundle to 2429? We have an updated lore timeline now and I've been meaning to change it to the year of first contact

File: code/datums/components/scope.dm [hunk 1]
```diff
@@ -105,36 +150,41 @@
...
  * * user: The mob we are starting zooming on.
 */
 /datum/component/scope/proc/zoom(mob/user)
 	if(isnull(user.client))
 		return
-	if(zooming)
+	if(HAS_TRAIT(user, TRAIT_USER_SCOPED))
+		user.balloon_alert(user, "already zoomed!")
 		return
-	user.client.mouse_override_icon = 'icons/effects/mouse_pointers/scope_hide.dmi'
```
Comment (Fikou) [inline code/datums/components/scope.dm:128]: 1. why
2. you didnt delete this dmi file

Comment (Ghommie) [reply code/datums/components/scope.dm:128]: 1. The mouse pointer being invisible sucks. The reticle or crosshair doesn't follow its position and stays fixed on the center of the screen, so you may as well accidentally move the cursor out of borders while navigating your view around.
2. I forgor :skull:

**Comment by JohnFulpWillard** [comment]
I don't know if I like binoculars using it, people using them have to know where to look rather than being able to just watch in a general direction, no?

**Comment by Ghommie** [comment]
> I don't know if I like binoculars using it, people using them have to know where to look rather than being able to just watch in a general direction, no?

It's actually let restrictive this way. Granted your screen view is unchanged, the fact it doesn't increase the view size means it's less laggy on client-side. It also allows you to move around and rotate a smooth 360° angle without disrupting the zoom.

**Comment by JohnFulpWillard** [comment]
we shall see fucker

**Comment by FieryMorty** [comment]
I will say, this is an amazing improvement over the old binoculars, which could break the screen occasionally until used again.

A really good PR!

However, I cannot see rune text of someone's name when I hover over them with my cursor now, and I can no longer examine someone from afar with the binoculars.

Is it possible to re-add that?

**Comment by Ghommie** [comment]
It's possible. I don't know if that'd classify as a qol or a fix though.

**Pull request merged**